test/upgrade/run-upgrade.sh

```diff
@@ -5,129 +5,140 @@ set -vex
+IFS=',' read -r -a availableVersions <<<"$AVAILABLEVERSIONS"
+len=${#availableVersions[@]}
+i=0
+while [ $i -lt $(($len - 1)) ]; do
+  echo "Upgrading from version ${availableVersions[$i]} to ${LATESTVERSION}"
+  # Check out the older version of the codebase and launch the engine with a local postgres
+  git checkout v${availableVersions[$i]}
+  yarn docker-compose &
+  yarn wait-on http://localhost:8088/health/
+
+  # Create a new database connected to a test account
+  curl http://localhost:8088/v1/db/connect/to_upgrade
   psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
-    INSERT INTO aws_account (access_key_id, secret_access_key, region)
-    VALUES ('${AWS_ACCESS_KEY_ID}', '${AWS_SECRET_ACCESS_KEY}', 'us-east-1');
-  ";
-else
-  psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
-    INSERT INTO aws_credentials (access_key_id, secret_access_key)
-    VALUES ('${AWS_ACCESS_KEY_ID}', '${AWS_SECRET_ACCESS_KEY}');
-  ";
-  psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
-    SELECT * FROM iasql_sync();
-  ";
-  psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
-    SELECT * FROM default_aws_region('us-east-1');
-  ";
-fi
-
-# Add one module that we can use to verify that everything actually comes back up
-psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
-  SELECT * FROM iasql_install('aws_ec2');
-";
-
-# Shut down the database and engine, switch back to the current commit, and fire up a new engine
-docker container stop $(basename ${PWD})_change_engine_1
-docker container stop $(basename ${PWD})_postgresql_1
-git checkout ${CURRENTGITSHA}
-yarn docker-compose &
-yarn wait-on http://localhost:8088/health/
-
-# Actually trigger the upgrade and loop until upgraded (or fail)
-psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
-  SELECT * FROM iasql_upgrade();
-";
-UPGRADECHECKCOUNT=30
-ISUPGRADED=false
-while [ ${UPGRADECHECKCOUNT} -gt 0 ]; do
-  AWSACCOUNTMODULE=`psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -AXqtc "
-    SELECT name FROM iasql_module WHERE name like 'aws_account%'
-  "`
-  if [ "${AWSACCOUNTMODULE}" == "aws_account@${LATESTVERSION}" ]; then
-    ISUPGRADED=true
-    UPGRADECHECKCOUNT=0
+      select iasql_install(
+        'aws_account'
+      );
+    "
+
+  # Determine which kind of 'aws_account' module this is (TODO: Remove this branch once v0.0.19 is oldest)
+  AWSACCOUNTTABLE=$(psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -AXqtc "
+      SELECT \"table\" FROM iasql_tables WHERE \"table\" = 'aws_account';
+    ")
+  if [ "${AWSACCOUNTTABLE}" == "aws_account" ]; then # It's the old style
+    psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
+        INSERT INTO aws_account (access_key_id, secret_access_key, region)
+        VALUES ('${AWS_ACCESS_KEY_ID}', '${AWS_SECRET_ACCESS_KEY}', 'us-east-1');
+      "
+  else
+    psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
```

Comment:
would make sense to store this connection string on a variable, it's repeated everywhere